crates/modelardb_server/tests/integration_test.rs

```diff
@@ -168,6 +168,68 @@ fn test_can_get_schema() {
     stop_modelardbd(flight_server);
 }
 
+#[test]
+#[serial]
+fn test_can_list_actions(){
+    let temp_dir = tempfile::tempdir().expect("Could not create a directory.");
+    let flight_server = start_modelardbd(temp_dir.path());
+
+    let runtime = Runtime::new().expect("Unable to initialize runtime.");
+    let mut flight_service_client = create_apache_arrow_flight_service_client(&runtime, HOST, PORT)
+        .expect("Cannot connect to flight service client.");
+
+    let mut actions = Vec::new();
+    runtime.block_on(async {
+        let mut response = flight_service_client
+            .list_actions(Request::new(arrow_flight::Empty {}))
+            .await
+            .expect("Could not retrieve actions.")
+            .into_inner();
+
+
+        while let Some(action) = response.next().await{
```

Comment:
`stream` implements [`collect()`](https://docs.rs/futures/latest/futures/stream/struct.Collect.html) so it should not be necessary to create a `mut Vec` as it should be possible to simply collect the result into an immutable `Vec` [like in `compressed_data_manager`](https://github.com/ModelarData/ModelarDB-RS/blob/master/crates/modelardb_server/src/storage/compressed_data_manager.rs#L267).

Reply:
Like this?

Reply:
Yes, but it is unclear why the variables and`response` and `actions` are introduced, maybe replace them with `flight_service_client.list_actions(...` followed by each operation on the resulting stream on a separate line. By replacing `.await;` with `.await` the result of the transformations will be returned from `block_on()`.

Reply:
Fixed.